subsys/net/lib/tls_credentials/Kconfig

```diff
@@ -21,6 +21,7 @@ choice TLS_CREDENTIALS_BACKEND
 
 config TLS_CREDENTIALS_BACKEND_VOLATILE
 	bool "TLS credentials management volatile backend"
+	depends on !TLS_CREDENTIAL_CSR
```

Comment:
should this dependency be moved to the `TLS_CREDENTIAL_CSR` option?
as in, `TLS_CREDENTIAL_CSR` depends on `!TLS_CREDENTIALS_BACKEND_VOLATILE`
or maybe better depends on `TLS_CREDENTIALS_BACKEND_PROTECTED_STORAGE`

Reply:
Personally, I think it makes more sense for the backends to specify which features they support, rather than for the features to specify which backends support them